components/src/core/header/headerNavigationIcon.tsx

```diff
@@ -13,6 +13,7 @@ const defaultStyles = StyleSheet.create({
         height: 40,
         width: 40,
         margin: -8,
+        marginTop: -16,
```

Comment:
Slapping negative margins on here feels like a hack? What's the underlying issue for the misalignment?
Looks to me like the icons are correctly centered, but the text is what's off:
![image](https://user-images.githubusercontent.com/29152776/107076880-180cc480-67ba-11eb-8e6e-dc746f70ae1c.png)

Reply:
The misalignment is being caused by the titleContainer style in the header using justifyContent: flex-end. When it's collapsed, the text is aligning at the bottom of the wrapper and overflowing over the top. This is not trivial to fix, since we need it to be flex end so that the text will stay pinned to the bottom when the header is expanded.

A (hacky) way to do this is to add a listener on the headerHeight animated value and then conditionally set the justify property based on whether it is collapsed or expanding.

```
const { headerHeight } = useHeaderHeight();
const [collapsed, setCollapsed] = useState(true);
useEffect(() => {
        const listener = headerHeight.addListener(({value}) => setCollapsed(value === REGULAR_HEIGHT))
        return () => {
            headerHeight.removeListener(listener);
        }
    }, [])
...
        <Animated.View
            style={[
                defaultStyles.titleContainer,
                {
                    justifyContent: collapsed ? 'center' : 'flex-end',
                },
...
```

Reply:
I'm fine to log that as a separate issue and deal with it in a different PR than the UserMenu update.